### PR TITLE
update the profiling C# code section of the doc for Godot 4

### PR DESCRIPTION
**Pull request by van800**
4.0

### CODE REVIEW

File: tutorials/scripting/c_sharp/c_sharp_basics.rst [hunk 1]
```diff
@@ -393,8 +393,9 @@ packages the next time it builds the project.
...
 
-- `Mono log profiler <https://www.mono-project.com/docs/debug+profile/profile/profiler/>`_
-  is available for Linux and macOS. Due to a Mono change, it does not work on
-  Windows currently.
-- External Mono profiler like `JetBrains dotTrace <https://www.jetbrains.com/profiler/>`_
-  can be used as described `here <https://github.com/godotengine/godot/pull/34382>`_.
+The following tools may be used for performance and memory profiling of your managed code:
+- JetBrains Rider with dotTrace/dotMemory plugin
+- Standalone JetBrains dotTrace/dotMemory
+- Visual Studio
```
Comment (raulsntos) [inline tutorials/scripting/c_sharp/c_sharp_basics.rst:399]: Other lists seem to always have a new line before. Also, each element should have a period at the end.

```suggestion

- JetBrains Rider with dotTrace/dotMemory plugin.
- Standalone JetBrains dotTrace/dotMemory.
- Visual Studio.
```

File: tutorials/scripting/c_sharp/c_sharp_basics.rst [hunk 2]
```diff
@@ -393,8 +393,9 @@ packages the next time it builds the project.
...
 ----------------------
 
-- `Mono log profiler <https://www.mono-project.com/docs/debug+profile/profile/profiler/>`_
-  is available for Linux and macOS. Due to a Mono change, it does not work on
-  Windows currently.
-- External Mono profiler like `JetBrains dotTrace <https://www.jetbrains.com/profiler/>`_
-  can be used as described `here <https://github.com/godotengine/godot/pull/34382>`_.
+The following tools may be used for performance and memory profiling of your managed code:
+- JetBrains Rider with dotTrace/dotMemory plugin
+- Standalone JetBrains dotTrace/dotMemory
```
Comment (raulsntos) [inline tutorials/scripting/c_sharp/c_sharp_basics.rst:398]: Consider re-adding the link to `https://www.jetbrains.com/profiler/`.

**Review by raulsntos — APPROVED** [review]
LGTM minus a few nitpicks. I don't use Rider or Visual Studio but I trust the information is correct.

File: tutorials/scripting/c_sharp/c_sharp_basics.rst [hunk 3]
```diff
@@ -393,8 +393,10 @@ packages the next time it builds the project.
...
-  Windows currently.
-- External Mono profiler like `JetBrains dotTrace <https://www.jetbrains.com/profiler/>`_
-  can be used as described `here <https://github.com/godotengine/godot/pull/34382>`_.
+The following tools may be used for performance and memory profiling of your managed code:
+
+- JetBrains Rider with dotTrace/dotMemory plugin.
+- Standalone JetBrains dotTrace/dotMemory.
+- Visual Studio.
+
+Profiling managed + unmanaged code at once is possible with both JetBrains tools and Visual Studio, but limited to Windows OS.
```
Comment (mhilbrunner) [inline tutorials/scripting/c_sharp/c_sharp_basics.rst:402]:
```suggestion
Profiling managed and unmanaged code at once is possible with both JetBrains tools and Visual Studio, but limited to Windows.
```

**Comment by mhilbrunner** [comment]
Thank you, merged!

**Pull request merged**